csharp/src/Apache.Arrow/Arrays/Time64Array.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or moreDate32Array
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Apache.Arrow.Types;
+using System;
+using System.IO;
+
+namespace Apache.Arrow
+{
+    /// <summary>
+    /// The <see cref="Time64Array"/> class holds an array of longs, where each value is
+    /// stored as the number of seconds/ milliseconds (depending on the Time64Type) since midnight.
+    /// </summary>
+    public class Time64Array : PrimitiveArray<long>
+    {
+        /// <summary>
+        /// The <see cref="Builder"/> class can be used to fluently build <see cref="Time64Array"/> objects.
+        /// </summary>
+        public class Builder : PrimitiveArrayBuilder<long, long, Time64Array, Builder>
```

Comment:
Does `Builder` need to inherit `PrimitiveArrayBuilder<TFrom, TTo, TArray, TBuilder>`?
Can we directly use `PrimitiveArrayBuilder<TFrom, TTo, TArray, TBuilder>` if `ConvertTo` does nothing?
It is similar for `Time32Array.Builder`

Reply:
I initially played around with using Timespan which needed this, but this is not required now. I have made this change